node/router/src/lib.rs

```diff
@@ -210,10 +213,12 @@ impl<N: Network> Router<N> {
         self.tcp.listening_addr().expect("The TCP listener is not enabled")
     }
 
-    /// Returns `true` if the given IP is this node.
-    pub fn is_local_ip(&self, ip: &SocketAddr) -> bool {
-        *ip == self.local_ip()
-            || (ip.ip().is_unspecified() || ip.ip().is_loopback()) && ip.port() == self.local_ip().port()
+    /// Returns `true` if the IP provided by a peer is a valid candidate peer address.
+    pub fn is_peer_ip_valid(&self, ip: &SocketAddr) -> bool {
+        *ip != self.local_ip()
+            && !ip.ip().is_unspecified()
+            && !ip.ip().is_loopback()
+            && if let IpAddr::V4(ip) = ip.ip() { !ip.is_broadcast() } else { true }
```

Comment:
so V6 addresses are always valid?

Reply:
These also have multicast.

Reply:
no, they just can't be broadcast, so the final condition is ignored for them

Reply:
> These also have multicast.

indeed, there are some quirks involved there, but I don't think we've seen any issues with them in the past